### PR TITLE
fix(router): re-export of Type

### LINKED ISSUE #3632: Router should not export Type Interface
We currently get two docs exported for Type Interface: one for core, and one for router. This is confusing -- and I think unnecessary duplication? Can we make Router depend on the Type exported from core?

### PR DESCRIPTION
**Pull request by mhevery**
Closes #3632

### CODE REVIEW

File: docs/docs-package/templates/type-definition.template.html [hunk 1]
```diff
@@ -1,72 +0,0 @@
-
```
Comment (mhevery) [inline docs/docs-package/templates/type-definition.template.html]: removing unused file.

**Comment by btford** [comment]
@mhevery Travis is red. Can you rebase and push again?

**Comment by mhevery** [comment]
@btford This is not a working protatype, but only steps 1 and 2 out of 3. See https://github.com/angular/angular/issues/3632 for details. I was hoping that someone else can solve step no 3

**Comment by btford** [comment]
LGTM. If CI passes, merge away.

**Comment by btford** [comment]
CI did not pass :(

**Pull request closed without merging**